lib/_emerge/BinpkgExtractorAsync.py

```diff
@@ -129,26 +120,18 @@ def _xpak_start(self):
         self.args = [
             self._shell_binary,
             "-c",
-            (
-                "cmd0=(head -c %d -- %s) cmd1=(%s) cmd2=(tar -xp %s -C %s -f -); "
-                + '"${cmd0[@]}" | "${cmd1[@]}" | "${cmd2[@]}"; '
-                + "p=(${PIPESTATUS[@]}) ; for i in {0..2}; do "
-                + "if [[ ${p[$i]} != 0 && ${p[$i]} != %d ]] ; then "
-                + 'echo command $(eval "echo \\"\'\\${cmd$i[*]}\'\\"") '
-                + "failed with status ${p[$i]} ; exit ${p[$i]} ; fi ; done; "
-                + "if [ ${p[$i]} != 0 ] ; then "
-                + 'echo command $(eval "echo \\"\'\\${cmd$i[*]}\'\\"") '
-                + "failed with status ${p[$i]} ; exit ${p[$i]} ; fi ; "
-                + "exit 0 ;"
-            )
-            % (
-                pkg_xpak.filestat.st_size - pkg_xpak.xpaksize,
-                portage._shell_quote(self.pkg_path),
-                decomp_cmd,
-                tar_options,
-                portage._shell_quote(self.image_dir),
-                128 + signal.SIGPIPE,
-            ),
+            f"cmd0=(head -c {pkg_xpak.filestat.st_size - pkg_xpak.xpaksize} -- {portage._shell_quote(self.pkg_path)}) "
```

Comment:
Some of the quoting here makes me a bit nervous but I don't have a better suggestion right now either.

Reply:
@arthurzam @mgorny any ideas?

Reply:
@thesamesam if we are nervous about quoting, we can use `shlex.quote`, but I see we are using `portage._shell_quote` in this line, so we should be fine.

Can you explain what makes you nervous (like the full code block)

Reply:
Also, since this here is a single long string, I would use something like:

```python
textwrap.dedent(
    """\
        cmds
    """
)
```

In my opinion we have less python quotes, so easier to see the bash code.

Reply:
IMO, that would make it more complex/hard to read since you have to ensure all `{` get translated to `{{` (and vice versa), and it would be harder to pick out which ones are substitutions and which ones aren't.

Reply:
Perhaps something in between the two suggestions, where consecutive fstrings are multi-line and consecutive plain strings are multi-line?

Reply:
Coming back to this, I think it's OK as it is.